cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/classloader/SparkClassRewriter.java

```diff
@@ -663,6 +669,78 @@ public void visitMethodInsn(int opcode, String owner, String name, String desc,
     return cw.toByteArray();
   }
 
+  /**
+   * Rewrites the temporary file name used for checkpoints from checkpointDir/temp to
+   * checkpointDir/temp-(current_timestamp_milliseconds).
+   *
+   * @param byteCodeStream {@link InputStream} for reading in the original bytecode.
+   * @return the rewritten bytecode
+   */
+  private byte[] rewriteTempFileNameForCheckpoint(InputStream byteCodeStream) throws IOException {
+    final Type hadoopPathType = Type.getType(org.apache.hadoop.fs.Path.class);
+    final Type stringType = Type.getType(java.lang.String.class);
+    final Type stringBuilderType = Type.getType(java.lang.StringBuilder.class);
+    ClassReader cr = new ClassReader(byteCodeStream);
+    ClassWriter cw = new ClassWriter(0);
+
+    cr.accept(new ClassVisitor(Opcodes.ASM5, cw) {
+      @Override
+      public MethodVisitor visitMethod(int access, String name, String desc, String signature, String[] exceptions) {
+
+        MethodVisitor mv = super.visitMethod(access, name, desc, signature, exceptions);
+
+        return new MethodVisitor(Opcodes.ASM5, mv) {
```

Comment:
For generating code, it's easier to use `GeneratorAdapter` instead of `MethodVisitor` directly.